components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt

```diff
@@ -469,6 +476,7 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
                     // If we have autocomplete text, the cursor is at the boundary between
                     // regular and autocomplete text. So regardless of which direction we
                     // are deleting, we should delete the autocomplete text first.
+                    restartInput()
```

Comment:
We recently removed exactly this restart to fix an issue. See #911 and f789c0653b67557bf6525817b6d9e902cdcf56de.

Reply:
What a dilemma!
Without the restarting, we're having issues when suggestion strip is off.

We have to find a way to convey solutions.

Reply:
@pocmo  What do you think about disabling this functionally only on Amazon devices?  Both sides could be happy.

The [Build class](https://developer.android.com/reference/android/os/Build#MANUFACTURER) can provide us with enough info to do that.

Reply:
I don't remember whether this problem only showed on Amazon devices. @mcomella Do you remember whether you were able to reproduce #911 on other devices too?

Reply:
#911 was created from https://github.com/mozilla-mobile-skunkworks/firefox-connect/issues/484 – iirc, this was only reproducible on the Echo Show.

Reply:
Should I proceed with this approach ?

Reply:
It may be our only option at this point. Let's document this decision in code and a test though. :)

Reply:
👍

Reply:
I opened https://github.com/mozilla-mobile/android-components/pull/1666 with the Echo Show fix.

I chose to go with the IME because an IME with the same ID should have similar behavior so it seemed less fragile than using the build information.